### PR TITLE
[detext] Adding multilabel classification

### PR DESCRIPTION
**Pull request by amberkaur1**
# Description

Added multi-label classification task to detext. Updated input label format to array (shape [num_classes], assuming one-hot array input), changed loss and activation functions, added precision/recall metrics. Passed unit tests on detect-dev mp while being backwards compatible. 

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## List all changes 
Please list all changes in the commit.
* Added MULTILABEL_CLASSIFICATION TaskType and fixed
* Updated input label format to array (shape [num_classes], assuming one-hot array input
* Changed loss and activation functions for multi-label classification
* Added precision/recall metrics

# Testing
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Passed unit tests on detect-dev mp while being backwards compatible. Also passed unit tests on li-detext, but full pipeline with real data yet to be confirmed

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/detext/train/metrics.py [hunk 1]
```diff
@@ -54,6 +54,34 @@ def update_state(self, labels, scores, sample_weight=None):
...
+    """ Metric for computing Precision given DeText format input """
+
+    def update_state(self, labels, scores, sample_weight=None):
+        """ Accumulates metric statistics
+
+        :param scores: Tensor Predicted scores as logits. Shape=[batch_size, max_group_size(1)] for ranking. For classification, shape=[batch_size]
+        :param labels: Tensor Labels. Shape=[batch_size, max_group_size(1)] for ranking. For binary classification, shape=[batch_size]
+        :param sample_weight: Sample weight. Check the inherited class method for more detail
+        """
+        probabilities = tf.round(tf.nn.sigmoid(scores))  # Convert to probabilities in range [0, 1]
```
Comment (wang-jia-rui) [inline src/detext/train/metrics.py:67]: After rounding the probabilities will become predictions, please change the variable name.

File: src/detext/train/metrics.py [hunk 2]
```diff
@@ -54,6 +54,34 @@ def update_state(self, labels, scores, sample_weight=None):
...
+        return super(PrecisionMetric, self).update_state(labels, probabilities, sample_weight)
```
Comment (wang-jia-rui) [inline src/detext/train/metrics.py:68]: Please also update the variable name here.

File: src/detext/train/metrics.py [hunk 3]
```diff
@@ -54,6 +54,34 @@ def update_state(self, labels, scores, sample_weight=None):
...
+    """ Metric for computing Recall given DeText format input """
+
+    def update_state(self, labels, scores, sample_weight=None):
+        """ Accumulates metric statistics
+
+        :param scores: Tensor Predicted scores as logits. Shape=[batch_size, num_classes] for multilabel classification.
+        :param labels: Tensor Labels. Shape=[batch_size, num_classes] for multilabel classification
+        :param sample_weight: Sample weight. Check the inherited class method for more detail
+        """
+        probabilities = tf.round(tf.nn.sigmoid(scores))  # Convert to probabilities in range [0, 1]
```
Comment (wang-jia-rui) [inline src/detext/train/metrics.py:81]: Please also update the variable name in RecallMetric

**Review by wang-jia-rui — CHANGES_REQUESTED** [review]
Thank you for your effort on multilabel classification! Please check my comments below.

**Comment by amberkaur1** [comment]
> Thank you for your effort on multilabel classification! Please check my comments below.
